File: classTypes4/c#/ReptilBase.cs
// reptilbase.cs
public abstract class ReptilBase : AnimalBase, IReptil{
    // boolean en java es bool en c#
    private bool tieneEscamas;

    public ReptilBase(string nombreCientifico, double pesoKg, bool tieneEscamas) 
        : base(nombreCientifico, pesoKg){
        this.tieneEscamas = tieneEscamas;
    }
    
    // implementacion concreta del metodo de la interfaz ireptil
    public void termoregular(){
        Console.WriteLine("el reptil esta tomando el sol para elevar su temperatura corporal");
    }

    // metodo abstracto para un movimiento especifico
    public abstract void tipodelocomocion();
}

File: classTypes4/java/src/ReptilBase.java
public abstract class ReptilBase extends AnimalBase implements Reptil {
    
    private boolean tieneEscamas;

    public ReptilBase(String nombreCientifico, double pesoKg, boolean tieneEscamas) {
        super(nombreCientifico, pesoKg);
        this.tieneEscamas = tieneEscamas;
    }
    
    // implementación concreta del método de la interfaz Reptil
    @Override
    public void termoregular() {
        System.out.println("el reptil esta tomando el sol para elevar su temperatura corporal");
    }

    // Método abstracto para un movimiento específico
    public abstract void tipoDeLocomocion();
}